### DISCUSSION TITLE
Read Bool as array

### DISCUSSION THREAD

**Discussion by sandysugu** [ottowayi/pycomm3#116] [Q&A]
Is there a way to read bool data type as array in SLCDriver?
I'm using MicroLogix 1100. I can read Integers and Float as array it works completely fine, but I can't do the same with bool.

Here is the code I used for Bool arrays.
```python
from pycomm3 import SLCDriver 

with SLCDriver('192.168.0.51', slot=0) as plc:     
    print(plc.read('B3:0/0{5}')) 
```
Instead of showing array values from B3:0/0 to B3:0/4
```
[True, False, True, False, True] 
```
it only displays as tag something like 
```
B3:0/0, True, B, None
```
Thanks .

**Comment by ottowayi** [comment]
Bools are handled a little differently than ints or floats, so there is probably a bug in the bool request code and the element count.  I have an 1100 that I plan on getting setup this afternoon, so I will try and look into it this weekend.   If not I will try to get it fixed this week.

**Reply by sandysugu** [reply]
Thanks @ottowayi
